src/ui/src/builder/sidebar/BuilderSidebarToolkit.vue

```diff
@@ -77,11 +118,45 @@ function showAutogen() {
 
 const wf = inject(injectionKeys.core);
 const wfbm = inject(injectionKeys.builderManager);
+const tracking = useWriterTracking(wf);
+const { removeComponentsSubtree } = useComponentActions(wf, wfbm, tracking);
 const { removeInsertionCandidacy } = useDragDropComponent(wf);
 const query = ref("");
 
+const isSharedBlueprintsEnabled = computed(
+	() =>
+		Array.isArray(wf.featureFlags.value) &&
+		wf.featureFlags.value.includes("shared_blueprints"),
+);
+
 const rootComponentId = wfbm.activeRootId;
 
+// Get shared blueprints directly from component tree
+const sharedBlueprintsFromTree = computed(() => {
+	if (!isSharedBlueprintsEnabled.value) return [];
+	const allBlueprints = wf.getComponents("blueprints_root", {
+		sortedByPosition: true,
+	});
+	return allBlueprints.filter((c) => c.content?.isSharedBlueprint === true);
+});
+
+// Block types that should not be available when editing a shared blueprint
+const RESTRICTED_BLOCKS_IN_SHARED_BLUEPRINT = new Set([
+	"blueprints_runblueprint",
+	"blueprints_apitrigger",
+	"blueprints_uieventtrigger",
+	"blueprints_crontrigger",
```

Comment:
Let's forbid shared blueprints relying on other shared blueprints, no ?

```suggestion
	"blueprints_crontrigger",
	"blueprints_shared",
```

Reply:
```suggestion
		// Don't show shared blueprints when editing a shared blueprint (no nesting for now)
		if (
			!isSharedBlueprintsEnabled.value ||
			isEditingSharedBlueprint.value
		) {
			return [];
		}
```

We already do that here by hiding the shared blueprints category when editing a shared blueprint